modules/simulation/src/main/resources/schemas/neurosciencegraph/simulation/tracefeatureextraction/v0.1.2.json

```diff
@@ -0,0 +1,64 @@
+{
+  "@context": [
+    "{{base}}/contexts/neurosciencegraph/core/schema/v0.1.0",
+    {
+      "this": "{{base}}/schemas/neurosciencegraph/simulation/tracefeatureextraction/v0.1.2/shapes/"
+    }
+  ],
+  "@type": "nxv:Schema",
+  "imports": [
+    "{{base}}/schemas/neurosciencegraph/commons/activity/v0.1.4"
+  ],
+  "shapes": [
+    {
+      "@id": "this:TraceFeatureExtractionShape",
+      "@type": "sh:NodeShape",
+      "label": "This shape is for recording the activity of feature extraction",
+      "nodekind": "sh:BlankNodeOrIRI",
+      "targetClass": "nsg:TraceFeatureExtraction",
+      "and": [
+        {
+          "node": "{{base}}/schemas/neurosciencegraph/commons/activity/v0.1.4/ActivityShape"
+        },
+        {
+          "property": [
+            {
+              "path": "prov:used",
+              "minCount": 2
+            },
+            {
+              "path": "prov:used",
+              "name": "Trace collection",
+              "description": "Used trace collections.",
+              "sh:qualifiedValueShape": {
+                "class": "nsg:TraceCollection"
+              },
+              "qualifiedValueShapesDisjoint": true,
+              "qualifiedMinCount": 1,
+              "qualifiedMaxCount": 1
+            },
+            {
+              "path": "prov:used",
+              "name": "BluePyEfe configuration",
+              "description": "The used BluePyEfe configuration file (s).",
+              "sh:qualifiedValueShape": {
+                "class": "nsg:Configuration"
+            },
+              "qualifiedValueShapesDisjoint": true,
+              "qualifiedMinCount": 1,
+              "qualifiedMaxCount": 1
+            },
+            {
+              "path": "prov:generated",
```

Comment:
Can you explain a little bit more why you removed the other generated entities.

Reply:
We already have BluePyEfeFeatures shape which is a composition of the protocols, features and thresholds. Breaking down that shape into ETypeFeatureProtocol, TraceFeature and Threshold would be inefficient IMHO.